### ISSUE TITLE
Unable to resolve var: cider.piggieback/wrap-cljs-repl

### ISSUE THREAD

**Issue by matthias-margush** [liquidz/vim-iced#73]
I'm hitting this error on some of my projects (but not others). I haven't looked into it any further, but if I discover anything, I'll let you know. I worked around it locally by removing the piggieback plugin (https://github.com/matthias-margush/vim-iced/pull/1/files), since I don't need the plugin at the moment.

```
჻ iced repl
OK: Leiningen project is detected
Warning: implicit hook found: lein-environ.plugin/hooks
Hooks are deprecated and will be removed in a future version.
Error loading cider.piggieback: java.lang.Exception: namespace 'cljs.closure' not found, compiling:(cljs/repl.cljc:9:1)
Exception in thread "main" java.lang.RuntimeException: Unable to resolve var: cider.piggieback/wrap-cljs-repl in this context, compiling:(/private/var/folders/bf/h482ks9x7n1_7_vd08v570vndt0j0_/T/form-init6580301070413941321.clj:1:5461)
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7010)
        at clojure.lang.Compiler.analyze(Compiler.java:6773)
        at clojure.lang.Compiler.analyze(Compiler.java:6729)
        at clojure.lang.Compiler$InvokeExpr.parse(Compiler.java:3881)
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7005)
        at clojure.lang.Compiler.analyze(Compiler.java:6773)
        at clojure.lang.Compiler.analyze(Compiler.java:6729)
        at clojure.lang.Compiler$InvokeExpr.parse(Compiler.java:3881)
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7005)
        at clojure.lang.Compiler.analyze(Compiler.java:6773)
        at clojure.lang.Compiler.access$300(Compiler.java:38)
        at clojure.lang.Compiler$LetExpr$Parser.parse(Compiler.java:6368)
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7003)
        at clojure.lang.Compiler.analyze(Compiler.java:6773)
        at clojure.lang.Compiler.analyze(Compiler.java:6729)
        at clojure.lang.Compiler$BodyExpr$Parser.parse(Compiler.java:6100)
        at clojure.lang.Compiler$FnMethod.parse(Compiler.java:5460)
        at clojure.lang.Compiler$FnExpr.parse(Compiler.java:4022)
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7001)
        at clojure.lang.Compiler.analyze(Compiler.java:6773)
        at clojure.lang.Compiler.eval(Compiler.java:7059)
        at clojure.lang.Compiler.eval(Compiler.java:7052)
        at clojure.lang.Compiler.eval(Compiler.java:7052)
        at clojure.lang.Compiler.load(Compiler.java:7514)
        at clojure.lang.Compiler.loadFile(Compiler.java:7452)
        at clojure.main$load_script.invokeStatic(main.clj:278)
        at clojure.main$init_opt.invokeStatic(main.clj:280)
        at clojure.main$init_opt.invoke(main.clj:280)
        at clojure.main$initialize.invokeStatic(main.clj:311)
        at clojure.main$null_opt.invokeStatic(main.clj:345)
        at clojure.main$null_opt.invoke(main.clj:342)
        at clojure.main$main.invokeStatic(main.clj:424)
        at clojure.main$main.doInvoke(main.clj:387)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.lang.Var.applyTo(Var.java:702)
        at clojure.main.main(main.java:37)
Caused by: java.lang.RuntimeException: Unable to resolve var: cider.piggieback/wrap-cljs-repl in this context
        at clojure.lang.Util.runtimeException(Util.java:221)
        at clojure.lang.Compiler$TheVarExpr$Parser.parse(Compiler.java:720)
        at clojure.lang.Compiler.analyzeSeq(Compiler.java:7003)
        ... 35 more
REPL server launch timed out.
```

**Comment by devth** [comment]
I have this problem too. Not sure if it's related but I just upgraded to java 11 and nrepl 0.5.3 and leiningen 2.8.3. It's my first time trying vim-iced though, so I didn't have it working on previous versions.

**Comment by liquidz** [comment]
@matthias-margush @devth Thank you for your reporting!
I confirmed the problem with following steps.

* Java 11
* Leiningen 2.8.3
* `lein new foo && cd foo && iced repl`

**Comment by liquidz** [comment]
piggieback `0.3.10` depends on clojurescript `1.9.946`.
https://github.com/nrepl/piggieback/blob/0.3.10/project.clj#L10

And clojurescript `1.9.946` has a problem with deprecated modules on Java9.
https://dev.clojure.org/jira/browse/CLJS-2377

This problem is fixed by clojurescript `1.10.238`.
https://github.com/clojure/clojurescript/blob/master/changes.md#110238

**Comment by liquidz** [comment]
@matthias-margush @devth We should change piggieback to be optional like CIDER.
https://docs.cider.mx/en/latest/clojurescript/#piggieback

I'll fix `iced` command, and `iced#nrepl#cljs` codes.

**Comment by devth** [comment]
Sounds good thanks 😄

**Comment by liquidz** [comment]
@matthias-margush @devth I'm working in `dev` branch, and I added `--with-cljs` option to `iced` command.
`iced` command will detect use of CLJS automatically, so we won't need to specify this option basically.

Could you test `dev` branch?

**Comment by devth** [comment]
@liquidz verified the fix on `dev` branch. Thanks!

**Comment by matthias-margush** [comment]
@liquidz Works here too!

**Comment by liquidz** [comment]
closed by #84
